pycheribuild/projects/disk_image.py

```diff
@@ -458,13 +476,33 @@ def add_all_files_in_dir(self, root_dir: Path):
 
     @property
     def is_x86(self):
-        return False
+        return self.crosscompile_target.is_any_x86()
 
     def run_mkimg(self, cmd: list, **kwargs):
         if not self.mkimg_cmd or not self.mkimg_cmd.exists():
             self.fatal("Missing mkimg command! Should be found in FreeBSD build dir (or set $MKIMG_CMD)")
         self.run_cmd([self.mkimg_cmd] + cmd, **kwargs)
 
+    @property
+    def include_efi_partition(self):
+        # TODO: Make this unconditional once all branches support EFI
+        if self.crosscompile_target.is_riscv(include_purecap=True):
+            return (self.rootfs_dir / "boot/loader.efi").exists()
+        return False
+
+    @property
+    def include_swap_partition(self):
+        return self.crosscompile_target.is_riscv(include_purecap=True)
```

Comment:
Even in the minimal images?  And why only RISC-V?

Reply:
Why only RISC-V: because that's what was already being done (added for FETT it not the nicest way)

Why minimal images: oops yes good point

Reply:
Would it make sense to just add swap for all images? It makes the raw images a bit bigger but the compressed size should stay the same?

Reply:
If it's added to all images I'd like a way to disable that as I have image size constraints and swap doesn't help when the disk image is loaded into memory before booting.

Reply:
I'm not convinced it should be there by default for RISC-V to be honest, IMO it should've only been there for FETT disk images.